nixstatsagent/plugins/redis_stat.py

```diff
@@ -0,0 +1,160 @@
+    "total_commands_processed": "total commands processed",
+    #"instantaneous_ops_per_sec": "instantaneous ops per sec",
+    "total_net_input_bytes": "total net input bytes",
+    "total_net_output_bytes": "total net output bytes",
+    #"instantaneous_input_kbps": "instantaneous input kbps",
+    #"instantaneous_output_kbps": "instantaneous output kbps",
+    #"rejected_connections": "rejected connections",
+    #"sync_full": "sync full",
+    #"sync_partial_ok": "sync partial ok",
+    #"sync_partial_err": "sync partial err",
+    "expired_keys": "expired keys",
+    "evicted_keys": "evicted keys",
+    "keyspace_hits": "keyspace hits",
+    "keyspace_misses": "keyspace misses",
+    #"pubsub_channels": "pubsub channels",
+    #"pubsub_patterns": "pubsub patterns",
+    #"latest_fork_usec": "latest fork usec",
+    #"migrate_cached_sockets": "migrate cached sockets",
+
+# Replication section
+    #"role": "role",
+    #"connected_slaves": "connected slaves",
+    #"master_repl_offset": "master repl offset",
+    #"repl_backlog_active": "repl backlog active",
+    #"repl_backlog_size": "repl backlog size",
+    #"repl_backlog_first_byte_offset": "repl backlog first byte offset",
+    #"repl_backlog_histlen": "repl backlog histlen",
+
+# CPU section
+    #"used_cpu_sys": "used cpu sys",
+    #"used_cpu_user": "used cpu user",
+    #"used_cpu_sys_children": "used cpu sys children",
+    #"used_cpu_user_children": "used cpu user children",    
+
+# Cluster section
+    "cluster_enabled": "cluster enabled"
+}
+
+class Plugin(plugins.BasePlugin):
+    __name__ = 'redis_stat'
+
+    def run(self, config):
+        data = {}
+        stats = None
+        try:
+            redis_host = (config.get(__name__, 'host'))
+        except:
+            redis_host = '127.0.0.1'
+        try:
+            redis_port = (config.get(__name__, 'port'))
+        except:
+            redis_port = '6379'
+        try:
+            redis_db = (config.get(__name__, 'db'))
+        except:
+            redis_db = '0'
+        try:
+            redis_password = (config.get(__name__, 'password'))
+        except:
+            redis_password = ''
+
+        try:
+            redis_connection = redis.StrictRedis(host=redis_host,port=redis_port,db=redis_db,password=redis_password)
+            stats = redis_connection.info()
+        except Exception as e:
+            data['status']=0
+            data['msg']='Connection Error'
+            if not stats:
+                return data
+
+        for name, value in stats.iteritems():
+            try:
+                if name in METRICS.keys() :
+                    if METRICS[name] in ['used_memory_peak_human', 'used_memory_human']:
+                        value = float(value[0:-1])
+                    data[METRICS[name]] = value
+            except (ValueError, TypeError) as e:
+                data[name] = value
+        return data
+
+        print(json.dumps(data, indent=4, sort_keys=True))
```

Comment:
No need to print here, to test output of a plugin you can run "nixstatsagent test redis_stat"